Issue 34/Setup Allure summaries mergefix

Nevedrov opened the pull request:
- Created allure summaries github actions workflow
- Refactored tests infrastructure architecture
- Improved test descriptions

Nevedrov committed:
refactor: Rename SuiteHierarchyProvider class

Nevedrov committed:
feat: add user and role entities repositories mock provider classes

Nevedrov committed:
refactor: move MockMVC setup boiler code to separate MockMvcProvider class

Nevedrov committed:
feat: Add TokenStoreServiceMockProvider class

Nevedrov committed:
refactor: move MockMvc object mapper setup boiler code into separate JacksonMapperProvider class

Nevedrov committed:
refactor: move all helper controllers provider classes into appropriate wrapper directory

Nevedrov committed:
chore: move API request wrapper classes to appropriate class and delete outdated legacy wrappers

Nevedrov committed:
chore: rename MockMvc package

Nevedrov committed:
fix: update JacksonMapperProvider package

Nevedrov committed:
chore: delete unnecessary UserProvider outdated legacy class

Nevedrov committed:
refactor: change UserProvider API by creating UserRoleProvider class which provides functionality for unit and integration test using user and role entities repositories

Nevedrov committed:
refactor: simplify controller tests setup

Nevedrov committed:
build: add docker-compose file for github actions tests setup

Nevedrov committed:
build: update application.properties

Nevedrov committed:
build: update test.application.properties

Nevedrov committed:
build: create github actions auto allure tests workflow

Nevedrov committed:
chore: remove unnecessary UserChangePasswordResponse schema

Nevedrov committed:
refactor: improve test infrastructure architecture by splitting and renaming all test helpers components into three packages:
 - Providers: stateless classes with only static methods
 - Properties: entities without methods which only stores data but don't have any functionality
 - Components: tests components which combines properties data with providers functionality for them and provide tests ability to conveniently and flexibly use common functionality on correct data

Nevedrov committed:
chore: remove legacy AuthProvider class

Nevedrov committed:
fix: use RefreshTokenToMap function for request body in ValidateRefresh to filter passing null refresh_token body field

Nevedrov committed:
chore: remove unnecessary Slf4j annotation and unnecessary imports

Nevedrov committed:
chore: improve tests descriptions

Nevedrov committed:
chore: clean application.properties

Nevedrov committed:
Merge branch 'development' of https://github.com/KUB-Education/KUB-Backend-CE into issue-34/setup-allure-summaries-mergefix
chore: Merge developement to test with renamed controller

Nevedrov committed:
chore: improve test application.properties fields names

Nevedrov committed:
chore: improve test application.properties fields names

Nevedrov committed:
chore: clean unused imports

Nevedrov committed:
build: add application.properties environment variables to github actions workflow

Nevedrov committed:
chore: improve github actions enviroment variables secrets names

Nevedrov committed:
chore: improve github actions enviroment variables secrets names

Nevedrov committed:
chore: improve github actions enviroment variables secrets names

Nevedrov committed:
chore: improve github actions enviroment variables secrets names

Nevedrov committed:
test: try fix github actions test by adding smtp host and port environment variables

Nevedrov committed:
test: debug application.properties on virtual server

Nevedrov committed:
test: update workflow

Nevedrov committed:
fix: fix workflow

Nevedrov committed:
fix: fix workflow fetch dependencies step

Nevedrov committed:
refactor: change application configuration method for properties via secrets.properties

Nevedrov committed:
fix: restore jdk_version environment variable

Nevedrov committed:
fix: fixed typo in secrets.tests.properties.example file name

PavloKilko commented on kub-backend-ce/docker-compose.yaml:
I'm not sure that is good idea to copy files from Infrastructure repository into backend repo.

1) Infrastructure will changing and in this case you need edit configuration in two places after every change.
2) If you really want to use docker-compose file, you need git clone it on the GH workflow stage.

PavloKilko commented on kub-backend-ce/config/secrets.tests.properties.example:
I'm not really understand purpose of this separate secrets.tests.properties.example file. You just created duplicate of already example file. Wanna change secret-key for correct hashing - just modify secrets.properties.example file.

Same goes for smtp.* fields.

PavloKilko reviewed (changes requested):
Overall, refactor code, so:
1) you will use same secretes.properties.example, because its already example and can be used for testing
2) remove docker-compose related files from this repo and modify a GH workflow so you will just git clone files from the KUB-Infrastructure repository.

Nevedrov committed:
refactor:
- delete docker-compose.yaml from backend repo
- move secrets.tests.properties.example content into  secrets.properties.example
- change workflow to work with docker-compose file from KUB-Infrastructure repository directly and use secrets.properties.example instead of secrets.tests.properties.example

Nevedrov committed:
refactor: - delete secrets.tests.properties.example from backend repo

Nevedrov committed:
chore: update secrets.properties.example
